aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2152,6 +2155,25 @@ impl VMValidator for AptosVM {
             }
         }
 
+        if !self.features.is_zkid_enabled() || !self.features.is_open_id_signature_enabled() {
+            if let Ok(authenticators) = aptos_types::zkid::get_zkid_authenticators(&transaction) {
+                for (_, sig) in authenticators {
```

Comment:
I don't have much context, but do we have a limit on number of authenticators? This code is executed unmetered, have to be careful with for loops.

Reply:
yes, theres MAX_ZK_ID_AUTHENTICATORS_ALLOWED in zkid_validation